spec/support/yaml_helper.rb

```diff
@@ -0,0 +1,35 @@
+module YAMLHelper
+  module_function
+
+  # from https://stackoverflow.com/a/55705853
+  def duplicate_keys(file_or_content)
+    yaml = file_or_content.is_a?(File) ? file_or_content.read : file_or_content
+    duplicate_keys = []
+
+    validator = ->(node, parent_path) do
+      if node.is_a?(Psych::Nodes::Mapping)
```

Comment:
Should we add `psych` to `Gemfile` so that we're not relying on a transitive dependency?

Reply:
The built-in ruby parser is psych these days, so I think it's fine as-is?

```
irb(main):001:0> defined?(Psych)
=> nil
irb(main):002:0> require 'yaml'
=> true
irb(main):003:0> defined?(Psych)
=> "constant"
```

Reply:
> The built-in ruby parser is psych these days, so I think it's fine as-is?
> 
> ```
> irb(main):001:0> defined?(Psych)
> => nil
> irb(main):002:0> require 'yaml'
> => true
> irb(main):003:0> defined?(Psych)
> => "constant"
> ```

Ah, I didn't know that, was primarily going by...

https://github.com/18F/identity-idp/blob/fca67c6ddb41b3768abc346eddf89314f70c8916/Gemfile#L110

Reply:
FWIW the other place we use it isn't the IDP, it's the [idp-config repo](https://github.com/18F/identity-idp-config/blob/main/spec/support/yaml_helper.rb) which doesn't have `psych` in its Gemfile either